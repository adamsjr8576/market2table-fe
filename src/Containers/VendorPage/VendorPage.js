import React from 'react';
import { useSelector } from 'react-redux';
import { withRouter } from 'react-router-dom';
import PropTypes from 'prop-types';
import './VendorPage.scss';
import { addSelectedMarket, addVendors } from '../../actions';
import { images } from "../../images/images";

export const VendorPage = ({ history }) => {
  const { selectedVendor, selectedMarket } = useSelector(state => state);
  const products = selectedVendor.products.map(product => {
    return (
      <article className='product-article' key={product.id}>
        <p>{product.name}: ${product.price}</p>
        <p>{product.description}</p>
      </article>
    )
  })

  const handleBackToMarket = () => {
      history.push(`/markets/${selectedMarket}/vendors/`)
  }

  return (
    <section className='section-market-page-container'>
      <button className='vendor-page-button' onClick={handleBackToMarket}>Back to Market Page</button>
      <div>
        <h2 className='vendor-page-vendor-name'>{selectedVendor.name}</h2>
        <h3>Description: </h3>
          <p>{selectedVendor.description}</p>
        <img className='vendor-img-vendor-page' src={selectedVendor.image_link}></img>
          <h3>Products: </h3>
            {products}
      </div>
    </section>
  )
}

export default withRouter(VendorPage);

VendorPage.propTypes = {
  history: PropTypes.object,
  selectedMarketId: PropTypes.string,
  selectedVendor: PropTypes.object
}
