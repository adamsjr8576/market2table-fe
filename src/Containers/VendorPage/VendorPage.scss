.section-vendor-page-container {
    margin: auto;
    width: 75%;
    height: 80%;
    display: flex;
    align-items: center;
    justify-content: space-around;
    overflow: auto;
    border: 3px solid #3B3D39;
    border-radius: 10px;
    background-color: #F2F9F9;
    border-right: 2px solid black;
    text-align: center;
    flex-wrap: wrap;
}

.vendor-info-page-button-div {
  width: 90%;
  display: flex;
  align-items: flex-start;
}

.vendor-info-page-div {
  width: 45%;
  overflow: auto;
  height: 80%;
}

.vendor-info-page-products-div {
  width: 45%;
  overflow: auto;
  height: 80%;
}

.vendor-img-vendor-page{
    height: 300px;
    width: 350px;
    border-radius: 5px;
}

.vendor-page-vendor-name{
    text-align: center;
    text-decoration: underline;
}

.product-article{
    margin: auto;
    border: 3px solid #5E7838;
    border-radius: 10px;
    margin: 10px 0;
    background-color: white;
}

.vendor-page-button {
    width: 15%;
    min-width: 155px;
    margin-top: 5px;
    text-align: center;
    cursor: pointer;
    line-height: 2px;
    font-size: 1em;
    padding: 10px;
    border: 2px solid #5E7838;
    border-radius: 10px;
}

.vendor-page-button:hover,
.vendor-page-button:focus {
  transform: scale(1.05);
}
