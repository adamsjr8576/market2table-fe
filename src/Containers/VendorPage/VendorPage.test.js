import React from 'react';
import { shallow } from 'enzyme';
import { VendorPage } from './VendorPage';

const mockState = {
  selectedVendor: {
    name: 'Rolling in the dough',
    description: 'just a little bakery rolling in some dough',
    image_link: 'rollthatdough.png',
    products: [{name: 'plain ol dough', price: '5', description: 'some good white bread', id: '2342'}]
  },
  vendors: [{name: 'Rolling in the dough', description: 'just a little bakery rolling in some dough', image_link: 'rollthatdough.png'}]
}

jest.mock("react-redux", () => ({
  useSelector: () => mockState
}));

describe('VendorPage', () => {

  let wrapper;
  let mockaddSelectedMarket = jest.fn();
  let mockaddVendors = jest.fn();

  it('should match the snapshot', () => {
    wrapper = shallow(<VendorPage
      addSelectedMarket={mockaddSelectedMarket}
      addVendors={mockaddVendors}
    />);

    expect(wrapper).toMatchSnapshot();
  });
});
