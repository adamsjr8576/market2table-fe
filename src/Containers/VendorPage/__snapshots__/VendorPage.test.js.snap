// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VendorPage should match the snapshot 1`] = `
<section
  className="section-vendor-page-container"
>
  <button
    className="vendor-page-button"
    onClick={[Function]}
  >
    Back to Vendors Page
  </button>
  <div>
    <h2
      className="vendor-page-vendor-name"
    >
      Rolling in the dough
    </h2>
    <h3>
      Description: 
    </h3>
    <p>
      just a little bakery rolling in some dough
    </p>
    <img
      className="vendor-img-vendor-page"
      src="rollthatdough.png"
    />
    <h3>
      Products: 
    </h3>
    <article
      className="product-article"
      key="2342"
    >
      <p>
        plain ol dough
        : $
        5
      </p>
      <p>
        some good white bread
      </p>
    </article>
  </div>
</section>
`;
